service/grails-app/services/org/olf/licenses/LicenseHousekeepingService.groovy

```diff
@@ -0,0 +1,38 @@
+package org.olf.licenses
+
+import grails.gorm.transactions.Transactional
+import org.olf.licenses.License
+import org.olf.licenses.LicenseAmendment
+import com.k_int.web.toolkit.refdata.RefdataValue
+
+/**
+ * This service works at the module level, it's often called without a tenant context.
+ */
+@Transactional
+public class LicenseHousekeepingService {
+
+  public void triggerHousekeeping() {
+    this.checkUnsetValues();
+  }
+
+
+  def checkUnsetValues() {
+        log.debug("EndDateSemanticsCleanup: Check for unset values")
+
+          int i = 0
+          License.findAllByEndDateSemanticsIsNull().each {lic ->
+            lic.endDateSemantics = RefdataValue.lookupOrCreate('endDateSemantics','Implicit')
+            lic.save()
+            i++
+          }
+          if (i>0) log.debug("Updated endDateSemantics for ${i} License(s), setting it to 'implicit'")
+
+          int j = 0
+          LicenseAmendment.findAllByEndDateSemanticsIsNull().each {la ->
```

Comment:
This is ok for now. If we hit performance issues in futue we may want to do something along the lines of [this](https://github.com/folio-org/mod-agreements/blob/35a47d5e74d0060483be27869517cb2d657e2cc5/service/grails-app/services/org/olf/SubscriptionAgreementCleanupService.groovy?plain=1#L40-L76) to batch fetch them, but I think we don't need to optimise this right now

Reply:
Actually this might be a problem for the first time we run the cleanup method, as we'll be attempting to drag in all the license amendments, and there could be thousands of them. This probably should either be a batch fetch or a single HQL statement like:

`(UPDATE all SET enddate.id = :implicit WHERE enddate=null), [implicit: RefdataValue.lookupOrCreate('endDateSemantics','Implicit')?.id]` or something